lib/ansible/module_utils/azure_rm_common.py

```diff
@@ -0,0 +1,635 @@
+        assigned to the object.
+
+        :param tags: object metadata tags
+        :return: bool, dict of tags
+        '''
+        if isinstance(tags, dict):
+            new_tags = copy.copy(tags)
+        else:
+            new_tags = dict()
+        changed = False
+        if self.module.params.get('tags'):
+            for key, value in self.module.params['tags'].iteritems():
+                if not (new_tags.get(key) and new_tags[key] == value):
+                    changed = True
+                    new_tags[key] = value
+        if changed:
+            self.log('CHANGED: updated tags')
+        return changed, new_tags
+
+    def update_tags(self, tags):
+        '''
+        Call from the module to update metadata tags. Returns tuple
+        with bool indicating if there was a change and dict of new
+        tags to assign to the object.
+
+        :param tags: metadata tags from the object
+        :return: bool, dict
+        '''
+        changed = False
+        updated, new_tags = self._tag_update(tags)
+        if updated:
+            changed = True
+
+        if self.module.params['purge_tags']:
+            purged, new_tags = self._tag_purge(new_tags)
+            if purged:
+                changed = True
+        return changed, new_tags
+
+    def exec_module(self):
+        '''
+        Execute module implementation.
+
+        :return: None
+        '''
+        res = self.exec_module_impl(**self.module.params)
+        self.module.exit_json(**res)
+
+    def get_resource_group(self, resource_group):
+        '''
+        Fetch a resource group.
+
+        :param resource_group: name of a resource group
+        :return: resource group object
+        '''
+        try:
+            return self.rm_client.resource_groups.get(resource_group)
+        except CloudError:
+            self.fail("Parameter error: resource group {0} not found".format(resource_group))
+        except Exception, exc:
+            self.fail("Error retrieving resource group {0} - {1}".format(resource_group, str(exc)))
+
+    def _get_profile(self, profile="default"):
+        path = expanduser("~")
+        path += "/.azure/credentials"
+        try:
+            config = ConfigParser.ConfigParser()
+            config.read(path)
+        except Exception, exc:
+            self.fail("Failed to access {0}. Check that the file exists and you have read "
+                      "access. {1}".format(path, str(exc)))
+        credentials = dict()
+        for key in AZURE_CREDENTIAL_ENV_MAPPING:
+            try:
+                credentials[key] = config.get(profile, key, raw=True)
+            except:
+                pass
+
+        if credentials.get('client_id') is not None or credentials.get('ad_user') is not None:
+            return credentials
+
+        return None
+
+    def _get_env_credentials(self):
+        env_credentials = dict()
+        for attribute, env_variable in AZURE_CREDENTIAL_ENV_MAPPING.iteritems():

```

Comment:
Another py3 porting note for 2.2 and beyond: if you don't need to use ```.iteritems()``` (because the number of entries would be small) just use ```.items()```.  If you do need to use iteritems because the items take up a lot of memory, in 2.2 we'll be adding ansible.module_utils.six which has an iteritems function to replace the dict method.